### DISCUSSION TITLE
Multiple IP on Panel for Rust Server causing problems

### DISCUSSION THREAD

**Discussion by Mxthic** [pterodactyl/panel#2849] [Support] [answered]
question. so my fqdn is set to my secondary ip which is what i want to use for all servers. it was all working well and coming up in the server list (rust). however i noticed that you could still connect using the other ip and server viewing sites like battlemetrics brought up the main ip which i do not want it to do. when i tried port fowarding it and set it to that ip it refuses to come up in the server list. this is for rust and does anyone by chance have a idea of the cause

**Comment by trenutoo** [comment, marked as answer]
This has been brought up multiple times, ex [#2704](https://github.com/pterodactyl/panel/issues/2704). You would either have configured the docker to use host mode in which case any benefits of docker would be lost and Minecraft proxies won't work or you can set up routing to steam servers for them to identify the correct IP for the steam server list.

Using host mode is the easiest way of resolving the issue when using multiple ips, the forwarding IP to steam servers is more complicated.

**Reply by Mxthic** [reply]
Is there any guides on routing to steam servers?

**Reply by Mxthic** [reply]
Also what benefits of Docker. Can you state the main losses

**Reply by ElijahRus250** [reply]
You can limit resources

**Reply by Mxthic** [reply]
I think I was still able to when i tried it. I don't get how switching docker from network_mode to host would make it unable to limit like ram, cpu, etc.

**Reply by trenutoo** [reply]
> I think I was still able to when i tried it. I don't get how switching docker from network_mode to host would make it unable to limit like ram, cpu, etc.

That won't be lost since you're still using Docker and its containers, you will, however, lose the network isolation with all servers being exposed to the internet. For example, Minecraft proxies such as Bungeecord will cease to offer any proxy protection.

Host mode is only recommended if you are hosting for personal use, if looking to offer it on a commercial level then I would suggest not to enable host mode personally. The game servers would be able to use ports and IPs that they are not allocated to.

The routing to steam servers is a pretty complicated network solution to provide a step-by-step guide because it depends on your stack. I personally have a firewall in between through which everything is routed.

**Reply by VibeGAMESNL** [reply]
Try this:

nano /etc/systemd/system/ptip.service

> [Unit]
> Description=Sets up NAT rules for docker
> After=docker.service
> Requires=docker.service
> 
> [Service]
> TimeoutStartSec=0
> Restart=always
> ExecStartPre=/usr/bin/docker pull quay.io/dreae/docker-external-ip
> ExecStart=/usr/bin/docker run --net=host --cap-add=NET_ADMIN --cap-add=NET_RAW --volume /var/run/docker.sock:/var/run/docker.sock --restart=always --name %n quay.io/dreae/docker-external-ip
> ExecStop=/usr/bin/docker stop %n
> ExecStopPost=/usr/bin/docker rm -f %n
> ExecReload=/usr/bin/docker restart %n
> 
> [Install]
> WantedBy=multi-user.target

systemctl enable --now ptip

**Reply by Mxthic** [reply]
I found something similar called pteroq.service is this what you are referring to @VibeGAMESNL

**Reply by Mxthic** [reply]
Or do you want me to make a file like that? @VibeGAMESNL

**Reply by parkervcp** [reply]
I have written about this before @VibeGAMESNL 

The following doesn't enable external ip addresses for pterodactyl containers.
`ExecStart=/usr/bin/docker run --net=host --cap-add=NET_ADMIN --cap-add=NET_RAW --volume /var/run/docker.sock:/var/run/docker.sock --restart=always --name %n quay.io/dreae/docker-external-ip` 

It literally starts a docker container in host network mode. Which is that changing the pterodactyl network to `host` also does. It also gives it full docker socket control. For all you know you have joined your system to a botnet.

**Reply by VibeGAMESNL** [reply]
Unless pterodactyl or docker will release a fix for this, I don't see another option for this specific issue. It's an option and it **works.** I searched the whole damn internet. 

Yes **this** container will run in host mode and if you look at the github page of dreae you will see it will only add some extra iptable rules. 

It's fair of @parkervcp to say it isn't super safe, so do you want to take the 'risk' using it? Your choice. It works and it's an option. But not recommended if your for example are a hosting company.

**Reply by VibeGAMESNL** [reply]
> I found something similar called pteroq.service is this what you are referring to @VibeGAMESNL

No pteroq.service is like the cronjob system of pterodactyl if i'm right

**Reply by Mxthic** [reply]
The reason I wanted to do this was because I didnt want players knowing the main IP as I was going to host this for the public on Rust. Anyone can play. Don't know if its a bigger risk doing host mode or using the main IP for the Servers as well as the panel. Docker needa get on there shit lmao
